server/bin/unittests

```diff
@@ -365,28 +351,18 @@ for testname in ${tests}; do
         continue
     fi
     let count=count+1
-    if [[ ${mode} == "auto" ]]; then
-        ${_tdir}/utils/run-unittest ${testname} ${cmd} &
-        pids[${testname}]=${!}
-    else
-        ${_tdir}/utils/run-unittest ${testname} ${cmd}
-        report_test_result ${testname}
-        if [[ ${?} -ne 0 ]]; then
-            let failures=${failures}+1
-        fi
-    fi
 done
 if [[ ${count} -eq 0 ]]; then
     printf "No tests run!\n" >&2
-    let failures=1
-elif [[ ${mode} == "auto" ]]; then
+    failures=1
+else
     for testname in ${tests}; do
-        wait ${pids[${testname}]} > /dev/null 2>&1
-        report_test_result ${testname}
-        if [[ ${?} -ne 0 ]]; then
-            let failures=${failures}+1
+        cmd=${cmds[${testname}]}
+        if [[ -n "${cmd}" ]]; then
+            echo "${testname} ${cmd}"
         fi
-    done
+    done | parallel -k ${jobs_arg} ${_tdir}/utils/run-unittest
+    failures=${?}
```

Comment:
I think that, if `$?` is non-zero, we should set `failures` to `1` rather than copying the exit value of `parallel`.